### DISCUSSION TITLE
Global player bug

### DISCUSSION THREAD

**Discussion by noahcousins** [katspaugh/wavesurfer.js#3523] [Q&A]
Hey everybody,

My team and I have been working on a global audio player using Wavesurfer in Next.js. We've just about finished it, but are stuck on a bug.

For example, lets say we have two sound IDs, (id-1, and id-2). When id-1 is played, the global wave player reflects the media element correctly. When id-2 is played, it correctly displays the new media element and peaks. However, **when playing id-1 again after changing to id-2, the id-1 waveform is not displayed, and the sound doesn't play** (including on the original Wavesurfer component the media element & peaks are derived from).

To troubleshoot, we've tried to use the empty method. We've also tried using a unique ID with the peaks as a string.

I'm sure there is a very simple fix we're missing. If anyone could assist, it would be greatly appreciated! Code below:

import { useEffect, useRef } from "react";
import WaveSurfer from "wavesurfer.js";

export default function GlobalWavePlayer({ paused, media }) {
  const wavesurferRef = useRef(null);

  const { media: audio, peaks } = media;

  const id = "global";

  const destroyWaveSurfer = () => {
    if (wavesurferRef.current) {
      wavesurferRef.current.destroy();
      wavesurferRef.current = null;
    }
  };

  useEffect(() => {
    destroyWaveSurfer();

    if (!wavesurferRef.current) {
      wavesurferRef.current = WaveSurfer.create({
        container: `#global`,
        normalize: true,
        hideScrollbar: true,
        barGap: 1,
        barMinHeight: 3,
        barWidth: 3,
        height: 54,
        media: audio,
        peaks: peaks,
        cursorWidth: 0,
        waveColor: "#737373",
        progressColor: "#fafafa",
        barRadius: 1.5,
      });
    }

    return () => {
      destroyWaveSurfer();
    };
  }, [audio]);

  useEffect(() => {
    if (wavesurferRef.current) {
      if (paused) {
        wavesurferRef.current.finish();
      } else {
        wavesurferRef.current.play();
      }
    }
  }, [paused]);

  return (

    <div className="w-full h-full relative">
      <div id={id} className="w-full h-full"></div>
    </div>

  );
}

**Comment by katspaugh** [comment]
I can’t say what’s wrong with your code without debugging your app. Try following this example more closely: https://wavesurfer.xyz/examples/?react-global-player.js
